pkg/tools/gen/template.go

```diff
@@ -128,6 +128,9 @@ func formatName(name string) string {
 	if _, ok := kclKeywords[name]; ok {
 		return fmt.Sprintf("$%s", name)
 	}
+	if strings.Contains(name, "-") {
```

Comment:
Perhaps a regular expression should be used here to determine whether it is a valid identifier expression rather than a simple string check, as it may also be in the form of `a.b`. cc @jakezhu9